frame/contracts/proc-macro/src/lib.rs

```diff
@@ -316,9 +323,27 @@ impl EnvDef {
 			.ok_or(err("Invalid environment definition, expected `mod` to be inlined."))?
 			.1;
 
+		let extract_fn = |i: &syn::Item| match i {
+			syn::Item::Fn(i_fn) => Some(i_fn.clone()),
+			_ => None,
+		};
```

Comment:
You could map into `ItemFn` and then change HostFn::try_from() to accept that instead of `syn::Item`.